templates/aws-tap-entrypoint-new-vpc.template.yaml

```diff
@@ -665,29 +687,67 @@ Resources:
         QSS3BucketRegion: !Ref QSS3BucketRegion
         QSS3KeyPrefix: !Ref QSS3KeyPrefix
 Outputs:
+  ClusterArch:
+    Description: >-
+      TAP cluster architecture
+    Value: !Ref TAPClusterArch
   LinuxBastionEIP:
     Description: >-
       The public IP address of the Linux bastion host / bootstrap instance.
     Value: !GetAtt ExistingVpcStack.Outputs.LinuxBastionEIP
   UbuntuBastionAZ:
-    Condition: Ubuntu
+    Condition: UbuntuSingleCluster
     Description: >-
       The Availability Zone that the Linux bastion host / bootstrap instance
       is deployed in.
     Value: !GetAtt ExistingVpcStack.Outputs.UbuntuBastionAZ
   UbuntuBastionId:
-    Condition: Ubuntu
+    Condition: UbuntuSingleCluster
     Description: >-
       The EC2 instance ID of the Linux bastion host / bootstrap instance.
     Value: !GetAtt ExistingVpcStack.Outputs.UbuntuBastionId
   UbuntuBastionPublicDnsName:
-    Condition: Ubuntu
+    Condition: UbuntuSingleCluster
     Description: >-
       The public DNS name of the Linux bastion host / bootstrap instance.
     Value: !GetAtt ExistingVpcStack.Outputs.UbuntuBastionPublicDnsName
+# multi ubuntu bastion outputs
+  UbuntuMultiBastionAZ:
+    Description: >-
+      The Availability Zone that the Ubuntu bastion host / bootstrap instance
+      is deployed in.
+    Value: !GetAtt ExistingVpcStack.Outputs.UbuntuMultiBastionAZ
+    Condition: UbuntuMultiCluster
+  UbuntuMultiBastionId:
+    Description: >-
+      The EC2 instance ID of the Ubuntu bastion host / bootstrap instance.
+    Value: !GetAtt ExistingVpcStack.Outputs.UbuntuMultiBastionId
+    Condition: UbuntuMultiCluster
+  UbuntuMultiBastionPublicDnsName:
+    Description: >-
+      The public DNS name of the Ubuntu bastion host / bootstrap instance.
+    Value: !GetAtt ExistingVpcStack.Outputs.UbuntuMultiBastionPublicDnsName
+    Condition: UbuntuMultiCluster
```

Comment:
in the bastion userdata there were specific references to the eks clusters that relied on outputs. if we are either creating a single cluster or multi cluster i dont see how we can use the same userdata block because youll get unresolved resource dependencies error

Reply:
i have found a solution to avoid using multicluster bastion